### ISSUE TITLE
feat: provide self-contained binary build

### ISSUE THREAD

**Issue by xynydev** [blue-build/cli#6]
- Local installation on any distro using a custom script similar to rustup
- Takes less space in CI/CD

**Comment by gmpinder** [comment]
Yes, when we get a workflow setup with cargo smart-release (the release tool I've been using), I'll be able to start packaging binaries in the releases.

**Comment by gmpinder** [comment]
Actually, if you want you can run inside of the image I build. I package everything you need in `registry.gitlab.com/wunker-bunker/blue-build:latest` and `registry.gitlab.com/wunker-bunker/blue-build:latest-alpine`.

**Comment by tulilirockz** [comment]
nix could also be used to redistribute this program to any distro, what do you all think?
